vrrpd/vrrp.c

```diff
@@ -341,17 +341,17 @@ void vrrp_check_start(struct vrrp_vrouter *vr)
 	/* Macvlan interface must have a link local */
 	start = start && connected_get_linklocal(r->mvl_ifp);
 	whynot =
-		(!start && !whynot) ? "No link local address configured" : NULL;
+		(!start && !whynot) ? "No link local address configured" : whynot;
 	/* Macvlan interface must have a v6 IP besides the link local */
 	start = start && (r->mvl_ifp->connected->count >= 2);
```

Comment:
This should use the new count function as well, I believe. Obviously very unlikely to arise in practice but lets us log an error in the situation a vrrp macvlan device has a v6 link local and a v4 address